src/Middleware/LanguageSwitcherMiddleware.php

```diff
@@ -148,6 +165,19 @@ private function __setCookieAndLocale($locale)
     }
 
     /**
+     * Loads additional config files that require the language to be set correctly.
+     *
+     * @return void
+     */
+    private function __loadConfigFiles()
+    {
+        $additionalConfigs = $this->getConfig('additionalConfigFiles');
```

Comment:
If you use `getConfig()` the minimum CakePHP requirement should be raised to `3.4`.